src/globus_sdk/services/transfer/client.py

```diff
@@ -392,6 +392,7 @@ def endpoint_search(
         filter_owner_id: str | None = None,
         filter_host_endpoint: UUIDLike | None = None,
         filter_non_functional: bool | None = None,
+        filter_entity_type: str | None = None,
```

Comment:
I would like us to annotate this with the `Literal` enumeration of valid strings.
I'll explain why below; first, the suggestion:
```suggestion
        filter_entity_type: Literal[
            "GCP_mapped_collection",
            "GCP_guest_collection",
            "GCSv5_endpoint",
            "GCSv5_mapped_collection",
            "GCSv5_guest_collection",
            "GCSv4_host",
            "GCSv4_share",
        ] | None = None,
```

There are a few reasons I have for preferring the `Literal` here.

With a literal annotation, at runtime the value is just a `str`. That means that if the service adds a new valid value which isn't reflected in the client, you can just pass `filter_entity_type="GCSv6_NewHotness",  # type: ignore` and it will work. In summary: "annotating as a `Literal` doesn't change the runtime type at all".

But `Literal` also provides
- documentation as rendered in our sphinx docs
- documentation as loaded/used by code completion and other analyzers (e.g., autocomplete)
- value checking/linting via type checkers